### PR TITLE
[NOREF] Update TruncatedText component to take line clamp prop

### PR DESCRIPTION
**Pull request by garyjzhao**
# [NOREF]

## Description

The `TruncatedText` component currently only takes a character limit. But recently, there have been designs asking to truncate text to _x_
 amount of lines (line clamping) instead of character count. This PR updates the `TruncatedText` component to take in line clamp value

This PR proposes moving away from using `MentionTextArea` to parse and render the text with all of its text entity and uses [html-react-parser](https://github.com/remarkablemark/html-react-parser) to parse and render said text

## How to test this change

1. `yarn install` && restart servers
2.  Navigate to a CR and TDL side panel and see that CR Summary is line clamped to 5, instead of a character count.
3. Look elsewhere in the app to ensure any other instances of `TruncatedText` still renders correct

## PR Author Checklist

- [ ] I have provided a detailed description of the changes in this PR.
- [ ] I have provided clear instructions on how to test the changes in this PR.
- [ ] I have updated tests or written new tests as appropriate in this PR.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.

### CODE REVIEW

File: src/components/TruncatedText/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      query: GetModelPlanBaseDocument,
+      variables: { id: 'f11eb129-2c80-4080-9440-439cbe1a286f' }
+    },
+    result: {
+      data: {
+        modelPlan: {}
+      }
+    }
+  }
+];
```
Comment (garyjzhao) [inline src/components/TruncatedText/index.test.tsx:20]: Test throw error if Provider (and therefore mocks) were not included in test. Is there another way to solve this? Currently, this is just a random mock and not used in `TruncatedText` component

Comment (patrickseguraoddball) [reply src/components/TruncatedText/index.test.tsx:20]: That's the incorrect query to mock.  TruncatedText utilized the `MentionTextArea`.  We might actually might want to not use this.  The `MetionTextArea` has a query embedded within it - `useSearchOktaUsersLazyQuery`.  I forgot why I decided to utilize `MentionTextArea`.  I don't think that was the right choice, considering how much overhead that component has built in.  We should probably try to build `TruncatedText` without it.  Sorry this was my fault

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]
I dont think you addressed that unit test in your PR.  It's mocking the incorrect query.  Also we can replace MentionTextArea if you want, would be a good time.  This would make us not need to mock the query

**Review by patrickseguraoddball — APPROVED** [review]
LG, nice!

**Pull request merged**